Clustering.Rmd

```diff
@@ -0,0 +1,621 @@
+
+train = subset(quant_data, sample == TRUE)
+test  = subset(quant_data, sample == FALSE)
+```
+
+```{r}
+library(klaR)
+help(kmodes)
+kmodes.result=kmodes(data=train,2)
+names(kmodes.result)
+```
+
+```{r}
+clustering.matrix=matrix(0,nrow=350001,ncol=4)
+colnames(clustering.matrix)=c("ClusterName","region_pickup","region_dropoff", "ride_type")
+clustering.matrix=as.data.frame(clustering.matrix)  
+clustering.matrix[,2:4]=train
+clustering.matrix[,"ClusterName"][kmodes.result$cluster==1]="ClusterOne"
+clustering.matrix[,"ClusterName"][kmodes.result$cluster==2]="ClusterTwo"
+clustering.matrix
+```
+
+
+```{r}
+ggplot(clustering.matrix, aes(ClusterName, fill = region_pickup)) + geom_bar()+
+labs(title = "Pickup region among different groups", x = "Cluster Name", y = "region_pickup")
+
+ggplot(clustering.matrix, aes(ClusterName, fill = region_dropoff)) + geom_bar()+
+labs(title = "Dropoff region among different groups", x = "Cluster Name", y = "region_dropoff")
+
+ggplot(clustering.matrix, aes(ClusterName, fill = ride_type)) + geom_bar()+
+labs(title = "Ride type among different groups", x = "Cluster Name", y = "ride_type")
+
+```
+
+Do k modes for rideshare type, weekend and tip_flag
+```{r}
+head(cleaned_data)
+```
+
+```{r}
+require(caTools)
+set.seed(232323)
+
+#select only numeric variables
+quant_data=cleaned_data[,c("flag_weekend", "tip_flag")]
+sample = sample.split(quant_data[,"flag_weekend"], SplitRatio = .7)
+
+train = subset(quant_data, sample == TRUE)
+test  = subset(quant_data, sample == FALSE)
+```
+
+```{r}
+library(klaR)
+kmodes.result=kmodes(data=train,2)
+names(kmodes.result)
+```
+
+```{r}
+clustering.matrix=matrix(0,nrow=350000,ncol=3)
+colnames(clustering.matrix)=c("ClusterName","flag_weekend", "tip_flag")
+clustering.matrix=as.data.frame(clustering.matrix)  
+clustering.matrix[,2:3]=train
+clustering.matrix[,"ClusterName"][kmodes.result$cluster==1]="ClusterOne"
+clustering.matrix[,"ClusterName"][kmodes.result$cluster==2]="ClusterTwo"
+clustering.matrix
+```
+
+```{r}
+library(ggplot2)
+cbbPalette <- c("#000000", "#E69F00")
+cbbPalette2<-c("#56B4E9", "#CC79A7")
+ggplot(clustering.matrix, aes(ClusterName, fill = flag_weekend)) + geom_bar()+
+labs(title = "Trips on weekend vs trips not on weekend between different groups", x = "Cluster Name", y = "flag_weekend")+
+scale_fill_manual(values=cbbPalette, labels=c("non-weekend", "weekend"))
+
+ggplot(clustering.matrix, aes(ClusterName, fill = tip_flag)) + geom_bar()+
+labs(title = "Tip vs nontip between different groups", x = "Cluster Name", y = "tip_flag")+
+scale_fill_manual(values=cbbPalette2, labels=c("non-tip", "tip"))
+
+
+```
+
+```{r}
+library(poLCA)
+f1=cbind(payment_type,speed_category_pickup_start,speed_category_dropoff_end,region_pickup,region_dropoff, ride_type)~1
+poLCA(f1,train,nclass=4,nrep=10,tol=.001,verbose=FALSE, graphs=TRUE)
+```
+
+## K nearest neighbor
+```{r}
+library(class)
+set.seed(232323)
+require(caTools)
+set.seed(232323)
```

Comment:
Seed already set up there :D

Reply:
Yeah I am having this seed again so I only need to run a specific chunk of codes each time